tests/cases/fourslash/quickInfoEvolvingArrayType.ts

```diff
@@ -0,0 +1,15 @@
+///<reference path="fourslash.ts" />
+
+// @strict: true
+// @checkJs: true
+// @allowJs: true
+// @outDir: ./dist
+
+// filename: index.js
+//// var C = [];
+//// for (var a in C)
+////     if (C.hasOwn/*1*/Property(a)) {
+////     }
+//// }
+
+verify.baselineQuickInfo();

```

Comment:
I'd strongly prefer that this be a compiler test over a fourslash test.

Reply:
Yeah, I had issues making it a real test, but I think my problem was actually not making it a JS test. Let me switch it.

Reply:
Yeah, it doesn't necessarily have to be JS. It can also be `noImplicitAny` I believe.

Reply:
Yeah, also noticed that (because I had a typo in the test when I copied the fourslash over and forgot to add the @ before "filename"). I can make it regular TS instead; it's probably cleaner that way.